FeatureModels/src/il/ac/openu/nlp/finalproject/models/featuremarker/FeatureMarkerModel.java

```diff
@@ -0,0 +1,49 @@
+package il.ac.openu.nlp.finalproject.models.featuremarker;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import il.ac.openu.nlp.finalproject.models.FeatureVector;
+import il.ac.openu.nlp.finalproject.models.MorphemeRecord;
+import il.ac.openu.nlp.finalproject.models.StructuredDataReader;
+import il.ac.openu.nlp.finalproject.models.TaggedFeatureVector;
+import il.ac.openu.nlp.finalproject.models.bagofwords.BagOfWordsModel;
+
+public class FeatureMarkerModel {
+	private static List<TaggedFeatureVector<String>> trainingData;
+	private static String ZERO_INDEX;
+	
+	public static List<TaggedFeatureVector<String>> buildFeatureMarkerTweetSize(Map<String, List<List<MorphemeRecord>>> userTweets, String ZERO_INDEX) throws IOException {
+		List<TaggedFeatureVector<String>> usersTweetsVector = new ArrayList<>();
+		FeatureMarker fm = new FeatureMarker("tweet_size", 10);
+
+		for (Map.Entry<String, List<List<MorphemeRecord>>> user : userTweets.entrySet()) {
+			for (List<MorphemeRecord> tweet : user.getValue()) {
+				FeatureVector<String> wordsPerTweet = new FeatureVector<String>(ZERO_INDEX);
+				
+				// Check if the tweet's length if equal or larger to the given length in the FeatureMarker
+				if (tweet.size() >= (int)fm.getValue()) {
+					// If yes, "re-build" the tweet itself since the variable 'tweet' contains discreet morphemes
+					for (MorphemeRecord morpheme : tweet) {
+						wordsPerTweet.put(morpheme.originalWord, 1.0);

```

Comment:
@ronenj : I think instead of thresholding the size and put 1 if the size > threshold, we can just put the actual size in the value of the feautre vector. Therefore, there is (maybe) no use for the feature marker class (If I understand clearly...)

Reply:
Also, the key to the wordsPerTweet map should be the name of the marker. in this case, "WordsPerTweet".
I think there is no need for the last for loop, because we are not interested in this case to go over all the morphemes. just count them...

Reply:
@asilev : 
a) Regarding first comment, I'm not placing '1' if the size if longer than the threshold. I'm adding the entire tweet to the list.
b) Regarding second comment: OK, will change the code.

Reply:
Let's talk tomorrow morning. I'm not sure I get the model...